JointMeasurements/ReferenceImplementation.qs

```diff
@@ -49,49 +50,47 @@ namespace Quantum.Kata.JointMeasurements {
         // The first state is a superposition of the states |++⟩ and |--⟩, 
         // which belong to the +1 eigenspace of the operator X ⊗ X;
         // the second one is a superposition of |+-⟩ and |-+⟩, which belong to the -1 eigenspace.
+        
         return Measure([PauliX, PauliX], qs) == Zero ? 0 | 1;
     }
     
     
     // Task 6*. Controlled X gate with |0⟩ target
     operation ControlledX_Reference (qs : Qubit[]) : Unit {
         H(qs[1]);
-        if (Measure([PauliZ, PauliZ], qs) == One) {
-            X(qs[1]);
-        }
+        
+        if (MeasureAllZ(qs) == One){
+            X(qs[1]);  
+		}
     }
     
     
     // Task 7**. Controlled X gate with arbitrary target
     operation ControlledX_General_Reference (qs : Qubit[]) : Unit {
         
-        body (...) {
```

Comment:
This is unfortunately one change we cannot do :-( This operation implements a unitary transformation, and has to have an adjoint variant defined. But at the same time it uses a measurement in it, so Q# compiler cannot compute its adjoint automatically (in general using a measurement renders an operation non-unitary, so it's reasonable behavior). So we need to use this syntax to define the body of the operation and its adjoint separately, rather than use `is Adj` to invoke automatic generation.

I undid this change, so now the CI build should pass.

By the way, thank you so much for trying this! I looked at the CI build failure and noticed that it fails in a different manner than I expected (both project build and notebook validation should fail, and only the latter does, the former passes), and thus found a recently introduced bug in our CI validation - which is great!

Reply:
This really sounds interesting. Great learning.
Feels good to be of help. 👍